q2_feature_table/_subsample.py

```diff
@@ -0,0 +1,39 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2017, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import biom
+
+
+def subsample(table: biom.Table, subsampling_depth: int,
+              axis: str) -> biom.Table:
+    if axis == 'feature':
+        # we are transposing the table due to biocore/biom-format#759
+        table = table.transpose()
+
+    if len(table.ids()) < subsampling_depth:
+        raise ValueError('The subsampling depth exceeds the number of '
+                         'elements on the desired axis. The maximum depth '
+                         'is: %d.' % len(table.ids()))
+
+    # the axis is always 'sample' due to the above transpose
+    table = table.subsample(subsampling_depth, axis='sample', by_id=True)
+
+    # the inverted axis is always observation due to the above transpose
+    invaxis = 'observation'
+    table.filter(lambda v, i, m: v.sum() > 0, axis=invaxis)
+
+    if axis == 'feature':
+        # reverse the transpose necessary due to biocore/biom-format#759
+        table = table.transpose()
+
+    if table.is_empty():
+        raise ValueError('The subsampled table contains no samples or '
+                         'features. Verify your table is valid and that you '
+                         'provided a shallow enough subsampling depth.')
```

Comment:
`contains either no samples or features` (since there's probably still one non-null axis, otherwise the subsampling_depth check would have tripped up first).

And then the second sentence seems a little backwards, wouldn't it be the shallow subsampling depths that are more likely to grab result in a null axis? Maybe explaining the `table.filter` call that happens would be a better hint as to what happens:

```python
"The subsampled table contains either no sample or features (samples/features that sum to zero are automatically removed)."